libnmstate/nm/applier.py

```diff
@@ -170,17 +170,18 @@ def set_ifaces_admin_state(ifaces_desired_state, con_profiles=()):
                 InterfaceState.ABSENT,
             ):
                 nmdevs = _get_affected_devices(iface_desired_state)
-                for nmdev in nmdevs:
-                    remove_devs_actions[nmdev] = [
+                for affected_nmdev in nmdevs:
+                    remove_devs_actions[affected_nmdev] = [
                         device.deactivate,
                         device.delete,
                     ]
-                    if nmdev.get_device_type() in (
-                        nmclient.NM.DeviceType.OVS_BRIDGE,
-                        nmclient.NM.DeviceType.OVS_PORT,
-                        nmclient.NM.DeviceType.OVS_INTERFACE,
-                    ):
-                        remove_devs_actions[nmdev].append(device.delete_device)
+                if (
+                    nmdev.is_software()
+                    and iface_desired_state[Interface.STATE]
+                    == InterfaceState.ABSENT
```

Comment:
The original code would remove OVS bridge, port and interface devices also if the state is down, not only when it is absent. Is this an intended change in the logic?

Reply:
Now that the veth has been explicitly checked, there is no need for this condition.

Reply:
Done